### ISSUE TITLE
Why values from PrelogitsFlatten are changing when fine-tune Inception V4 model?

### ISSUE THREAD

**Issue by gyang274** [tensorflow/models#2132]
My objective is to fine-tune Inception V4 on a new set of classes for prediction, and then also extract the next-to-last layer (prelogits) as features for searching.

I followed first the script slim/train_image_classifier.py, and specified `--checkpoint_exclude_scopes=InceptionV4/Logits,InceptionV4/AuxLogits --trainable_scopes=InceptionV4/Logits,InceptionV4/AuxLogits` for fine-tuning.

I then wrote a script to extract `'InceptionV4/Logits/PreLogitsFlatten/Reshape:0'` for each image.

My understanding is that the above prelogits are coming from average pooling of mixed7d/concat, and should not change in this fine-tune circumstance. In particular, I expected only the weights (1536 x num_classes) and bias (num_classes) that connects prelogits to logits will be trained. However, I found that the value of prelogits are changing as the model trained.

Would you kindly advice where I didn't fully understand the structure, or the fine-tuning script?

Thank you very much.

**Comment by shivaniag** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
